### PR TITLE
Fix potential loss of metrics when restart confignode

### PR DESCRIPTION
**Pull request by SpriCoder**
1. Move add metrics from init to leader ready
2. Move the management of replication factor metric into PartitionMetrics

### CODE REVIEW

File: iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/manager/schema/ClusterSchemaManager.java [hunk 1]
```diff
@@ -261,25 +240,11 @@ public TSStatus alterDatabase(DatabaseSchemaPlan databaseSchemaPlan) {
...
-      metricService
-          .getOrCreateGauge(
-              Metric.REPLICATION_FACTOR.toString(),
-              MetricLevel.CORE,
-              Tag.TYPE.toString(),
-              SCHEMA,
-              Tag.DATABASE.toString(),
-              databaseSchemaPlan.getSchema().getName())
-          .set(databaseSchemaPlan.getSchema().schemaReplicationFactor);
+      PartitionMetrics.bindDatabaseReplicationFactorMetrics(
```
Comment (OneSizeFitsQuorum) [inline iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/manager/schema/ClusterSchemaManager.java:243]: If changing the number of replica isn't currently supported, maybe we shouldn't do it here

File: iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/manager/partition/PartitionMetrics.java [hunk 2]
```diff
@@ -274,23 +326,11 @@ public static void bindDatabasePartitionMetrics(
         database,
         Tag.TYPE.toString(),
         TConsensusGroupType.DataRegion.toString());
+    bindDatabaseReplicationFactorMetrics(
```
Comment (OneSizeFitsQuorum) [inline iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/manager/partition/PartitionMetrics.java:329]: If put this line in it, we should revise the function bindDatabasePartitionMetricsWhenUpdate name?

File: iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/manager/schema/ClusterSchemaManager.java (hunk 1 above)
Comment (SpriCoder) [reply iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/manager/schema/ClusterSchemaManager.java:243]: But maybe I think we need to do that here?

**Review by OneSizeFitsQuorum — APPROVED** [review]
LGTM~
Awesome work!

**Pull request merged**